log10/load.py

```diff
@@ -506,6 +545,17 @@ def _init_log_row(func, *args, **kwargs):
                         else:
                             new_content.append(c)
                     m["content"] = new_content
+        if "tools" in kwargs_copy:
+            for t in kwargs_copy["tools"]:
+                new_function = {
+                    "name": t["name"],
```

Comment:
Also quick one, `t.get("name", None)` will be more robust to missing field types. Say, description etc. isn't available